STRF-10485 Create troubleshooting wiki for old themes (#1873)
bigcommerce/dev-docs: docs/stencil-docs/installing-stencil-cli/troubleshooting-your-setup.mdx

7. Verify your theme's launch at: http://localhost:3000.

> ### old version of `stencil-cli` included in package.json

Reviewer: ```suggestion
### The old version of `stencil-cli` included in package.json
```

> ### The old version of `stencil-cli` included in package.json

Reviewer 2: @jairo-bc that sounds fine

> ### The old version of `stencil-cli` included in package.json

Author: ```suggestion
### When Stencil CLI  is included in package.json
```

npm ERR! gyp ERR! stack                                ^
npm ERR! gyp ERR! stack SyntaxError: invalid syntax

```

It's because you have an old version of stencil CLI included in package.json. In general, it shouldn't be part of package.json (neither part of dev dependencies).

### Running `stencil bundle` errors

Nowadays, `stencil bundle` is running several validation checks on theme before it can be bundled and then pushed to the store. 
You may see those errors, because your theme had some missing properties in translations, frontmatter, that become required.

For example, running on the theme `stencil bundle` may occur such error

```text showLineNumbers copy
Error: Your theme's schemaTranslations.json has errors:
missing translation key "i18n.HeaderAndFooter"
```

, which indicates that `schemaTranslations.json` has missing translation key `i18n.HeaderAndFooter`.

```text showLineNumbers copy
Error: Missed schemaTranslations.json file
```

```text showLineNumbers copy
Error: Your theme's schemaTranslations.json has errors:
unused translation key "i18n.ProductSaleBadges"
unused translation key "i18n.ShowProductSaleBadges"
```

> meaning that there are some translation keys in the `schemaTranslations.json` file, that are not used in the theme, because are missed in `schema.json`

Reviewer 2: ```suggestion
There are some translation keys in the `schemaTranslations.json` file that are not used in the theme because they are missed in `schema.json`.
```

```test showLineNumbers copy
Error: Found unallowed trailing symbol in: "4,", while parsing frontmatter at ".....templates/pages/home.html".
```

> meaning that there is a trailing comma in the frontmatter of the `home.html` file.

Reviewer 2: ```suggestion
This means there is a trailing comma in the frontmatter of the `home.html` file.
```

> #### Node 16+ support 

Reviewer: ```suggestion
#### Node 18 support 
```
Let's support a single version officially: 18

> Just a note, that theme packages (webpack, e.g) might be updated to get latest Node versions support.

Reviewer 2: ```suggestion
Theme packages (webpack, e.g) might be updated to get the latest Node versions support.
```
